server/services/rescue.js

```diff
@@ -0,0 +1,153 @@
+const bunyan = require('bunyan');
+
+const log = bunyan.createLogger({ name: 'receive-service' });
+
+const path = require('path');
+
+const Response = require(path.join(__dirname, './response'));
+
+function isValidContactNumber(contactNumber) {
+  const regExpContactNumber = /^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$/;
+  return regExpContactNumber.test(contactNumber);
+}
+
+function isValidName(name) {
+  const regExpName = /^[A-Za-z\s]+$/;
+  return regExpName.test(name);
+}
+
+module.exports = class Rescue {
+  constructor(user, webhookEvent) {
+    this.user = user;
+    this.webhookEvent = webhookEvent;
+  }
+
+  handleStep() {
+    log.info('this.user- ', this.user);
+    let response = Response.genStandardErrorMessage();
+    const message = this.webhookEvent.message.text;
+    const { step } = this.user;
+    let locationInfo = '';
+
+    switch (step) {
+      case 'START':
+        response = Response.genStartMessage();
+        break;
+
+      case 'CONTACT_NAME_STEP':
+        if (isValidName(message) && message.length > 2) {
+          this.user.data = { contactPerson: message };
+          this.user.step = 'CONTACT_NUMBER_STEP';
+          response = Response.genText(
+            'Okay, please enter a contact number. \nSample format: (0917123123)',
+          );
+        } else {
+          response = Response.genText('Please enter a valid name.');
+        }
+        break;
+
+      case 'CONTACT_NUMBER_STEP':
+        if (isValidContactNumber(message)) {
+          this.user.data.contactNumber = message;
+          this.user.step = 'LOCATION_STEP';
+          response = Response.genText(
+            'Okay, please enter the full address or send a location pin.',
+          );
+        } else {
+          response = Response.genText('Please enter a valid contact number.');
+        }
+        break;
+
+      case 'LOCATION_STEP':
+        if (message.length > 9) {
```

Comment:
What is the basis for the message length > 9 for valid address

Reply:
I was thinking the length should be greater than <cityName> + 'city'. What length do you suggest? I will tune this down to `< 7` since Cebu City seems like the shortest city name.

Reply:
I think as long as it's not empty, it should be fine. We're not 100% sure what would be the shortest possible for an address to be valid